core/src/main/java/io/opencensus/stats/View.java

```diff
@@ -85,77 +114,77 @@ public static Name create(String name) {
     }
   }
 
-  /**
-   * A {@link View} for distribution-base aggregations.
-   */
+  /** The time window for a {@code View}. */
   @Immutable
-  @AutoValue
-  public abstract static class DistributionView extends View {
-    /**
-     * Constructs a new {@link DistributionView}.
-     */
-    public static DistributionView create(
-        Name name,
-        String description,
-        Measure measure,
-        DistributionAggregation distributionAggregation,
-        List<TagKey> tagKeys) {
-      return new AutoValue_View_DistributionView(
-          name,
-          description,
-          measure,
-          Collections.unmodifiableList(new ArrayList<TagKey>(tagKeys)),
-          distributionAggregation);
-    }
+  public abstract static class Window {
 
-    /**
-     * The {@link DistributionAggregation} associated with this
-     * {@link DistributionView}.
-     */
-    public abstract DistributionAggregation getDistributionAggregation();
-
-    @Override
-    public <T> T match(
-        Function<? super DistributionView, T> p0,
-        Function<? super IntervalView, T> p1) {
-      return p0.apply(this);
-    }
-  }
+    private Window() {}
 
-  /**
-   * A {@link View} for interval-based aggregations.
-   */
-  @Immutable
-  @AutoValue
-  public abstract static class IntervalView extends View {
     /**
-     * Constructs a new {@link IntervalView}.
+     * Applies the given match function to the underlying data type.
      */
-    public static IntervalView create(
-        Name name,
-        String description,
-        Measure measure,
-        IntervalAggregation intervalAggregation,
-        List<TagKey> tagKeys) {
-      return new AutoValue_View_IntervalView(
-          name,
-          description,
-          measure,
-          Collections.unmodifiableList(new ArrayList<TagKey>(tagKeys)),
-          intervalAggregation);
+    public abstract <T> T match(
+        Function<? super CumulativeWindow, T> p0,
+        Function<? super IntervalWindow, T> p1,
+        Function<? super Window, T> defaultFunction);
+
+    /** Cumulative (infinite interval) time {@code Window}. */
+    @Immutable
+    @AutoValue
+    public abstract static class CumulativeWindow extends Window {
+
+      CumulativeWindow() {}
+
+      /**
+       * Constructs a cumulative {@code Window} that does not have an explicit {@code Duration}.
+       * Instead, cumulative {@code Window} always has an interval of infinite {@code Duration}.
+       *
+       * @return a cumulative {@code Window}.
+       */
+      public static CumulativeWindow create() {
```

Comment:
Sounds reasonable. I think the naming is fine since having a constant is kind of implementation detail.